EIPS/eip-5114.md

```diff
@@ -0,0 +1,64 @@
+---
+eip: 5114
+title: Soulbound Token
+description: A token that is attached to a "soul" at mint time and cannot be transferred after that.
+author: Micah Zoltu (@MicahZoltu)
+discussions-to: https://ethereum-magicians.org/t/eip-5114-soulbound-token/9417
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-05-30
+requires: 721
+---
+
+## Abstract
+A soulbound token is a token that is bound to another ERC-721 token when it is minted, and cannot be transferred/moved after that.
+
+## Specification
+```solidity
+interface IERCXXXX {
+	// fired anytime a new instance of this token is minted
+	// this event **MUST NOT** be fired twice for the same `tokenId`
+	event Mint(uint256 indexed tokenId, address indexed erc721Token, uint256 indexed erc721TokenId);
+
+	// returns the ERC-721 token that owns this token.
+	// this function **MUST** throw if the token hasn't been minted yet
+	// this function **MUST** always return the same result every time it is called after it has been minted
+	// this function **MUST** return the same value as found in the original `Mint` event for the token
+	function ownerOf(uint256 index) external view returns (address erc721Token, uint256 erc721TokenId);
+	
+	// returns a censorship resistant URI with details about this token collection
+	// the metadata returned by this is merged with the metata return by `tokenUri(uint256)`
+	// the tokenUri **MUST** be immutable and content addressable (e.g., ipfs://)
+	// the tokenUri **MUST NOT** point at mutable/censorable content (e.g., https://)
+	// data from `tokenUri` takes precedent over data returned by this method
+	// any external links referenced by the content at `tokenUri` also **MUST** follow all of the above rules
+	function collectionUri() external view returns (string tokenUri);
+	
+	// returns a censorship resistant URI with details about this token instance
+	// the tokenUri **MUST** be immutable and content addressable (e.g., ipfs://)
+	// the tokenUri **MUST NOT** point at mutable/censorable content (e.g., https://)
+	// data from this takes precedent over data returned by `collectionUri`
+	// any external links referenced by the content at `tokenUri` also **MUST** follow all of the above rules
+	function tokenUri(uint256 tokenId) external view returns (string tokenUri);
```

Comment:
Wouldn't using `tokenURI` here be better than `tokenUri` as then the bytes4 signature of the function address in Solidity is equal to e.g. that of EIP-721?

Reply:
I considered it, but I don't think 721 compatibility matters much because the `ownerOf` signature is different and owner lookups require custom code, so this token won't natively work with existing ERC-721 tooling.

Perhaps there are specific tools that don't need to do owner lookups but do need to URI lookups that would benefit from consistency across methods?

*Personally* I would like to see a compelling reason to break the superior (IMO) naming convention of capitalizing only the first letter of acronyms (it avoids situations like `XMLHTTPRPC` with `XmlHttpRpc`).

Reply:
IMO naming convention arguments should be out of scope of this EIP. But a separate EIP could make normative statements on function capitalization conventions.

I way mainly arguing for consistency with EIP-721 but it's not a hard objection from my side.

Reply:
We've talked to some developers, and I'm guessing the logic is beyond many people's comprehension. But if you think deeply about some social phenomena, it is the only feasible solution to the problem of SBTs.

Reply:
New news, we have improved ERC721QS-1 and proposed ERC721QS-1P, which can also be used for NFT anti-theft, NFT leasing, NFT lending and designing SBTs! We even found cases where a similar approach was already used. We will update the documentation as soon as possible. But ERC721QS-2 saves gas more.